### PR TITLE
Add rule to check for AWS EC2 IMDSv1 

### PR DESCRIPTION
**Pull request by x64-latacora**

### CODE REVIEW

**Comment by x64-latacora** [comment]
@clintgibler :eyes:

**Comment by x64-latacora** [comment]
@clintgibler :eyes:

File: terraform/lang/security/ec2-imdsv1-optional.tf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# ruleid: ec2-imdsv1-optional
+resource "aws_instance" "test-instance" {
+  ami = "ami-0d5eff06f840b45e9"
+  metadata_options {
+    http_endpoint = "enabled"
+    http_tokens = "optional"
```
Comment (mschwager) [inline terraform/lang/security/ec2-imdsv1-optional.tf:6]: I think you'll need to move the `# ruleid: ...` comment above this line.

Also, can you add a true negative test as well? Similarly you can use the [`# ok: ...` syntax](https://semgrep.dev/docs/writing-rules/testing-rules/).

File: terraform/lang/security/ec2-imdsv1-optional.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+      AWS EC2 Instance allowing use of the IMDSv1
+    metadata:
+      references:
+        - https://aws.amazon.com/blogs/security/defense-in-depth-open-firewalls-reverse-proxies-ssrf-vulnerabilities-ec2-instance-metadata-service
+        - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#metadata-options
+    paths:
+      include:
+        - '*.tf'
+    patterns:
+      - pattern-inside: metadata_options ...
```
Comment (mschwager) [inline terraform/lang/security/ec2-imdsv1-optional.yaml]: I think the following may make more sense:

```
- pattern-inside: |
    metadata_options {
        ...
        ...
        ...
    }
```

Comment (x64-latacora) [reply terraform/lang/security/ec2-imdsv1-optional.yaml]: Improved the constraint.

**Comment by mschwager** [comment]
`pre-commit` is being extra finicky! (Especially with this new Github CI approval workflow :/)

After adding in the trailing new lines (but not an extra line at the end) we should be good to go!

**Comment by x64-latacora** [comment]
> After adding in the trailing new lines (but not an extra line at the end) we should be good to go!

:crossed_fingers:

**Review by mschwager — APPROVED** [review]
Nice! Thanks again!

**Pull request merged**